lib/SILGen/SILGenStmt.cpp

```diff
@@ -74,7 +74,9 @@ void SILGenFunction::eraseBasicBlock(SILBasicBlock *block) {
   assert(block->pred_empty() && "erasing block with predecessors");
   assert(block->empty() && "erasing block with content");
   if (block == StartOfPostmatter) {
-    StartOfPostmatter = &*std::next(block->getIterator());
+    StartOfPostmatter = (std::next(block->getIterator()) == F.end())
+                            ? nullptr
+                            : &*std::next(block->getIterator());
```

Comment:
Perhaps `StartOfPostmatter` could be made a `SILFunction::iterator` so that the `&*` becomes unnecessary. By a quick grep, it looks like we only ever check it for validity, advance it as an iterator anyway, or compare whether it refers to a specific basic block, all of which we ought to be able to do using the iterator representation.